tests/modes/apps/extensions/test_extension_remote.py

```diff
@@ -17,172 +16,84 @@ class TestExtensionRemote:
-
-        assert result == UrlParseResult(
-            use_git=False,  # don't use git for github, gitlab, codeberg URLs
-            url="https://github.com/user/repo",
-            path="user/repo",
-            protocol="https",
-            host="github.com",
-        )
-
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    @patch("ulauncher.modes.extensions.extension_remote.isdir")
-    def test_local_file_path(self, mock_isdir: MagicMock, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        mock_isdir.return_value = True
-        result = parse_extension_url("/local/path/to/extension")
-
-        assert result == UrlParseResult(
-            use_git=True,
-            url="file:///local/path/to/extension",
-            path="local/path/to/extension",
-            protocol="file",
-            host="",
-        )
-
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_github_url_sanitization(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        result = parse_extension_url("https://github.com/user/repo/blob/master")
-
-        assert result == UrlParseResult(
-            use_git=False, url="https://github.com/user/repo", path="user/repo", protocol="https", host="github.com"
-        )
-
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_gitlab_url_sanitization(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        result = parse_extension_url("https://gitlab.com/user/repo/issues")
-
-        assert result == UrlParseResult(
-            use_git=False, url="https://gitlab.com/user/repo", path="user/repo", protocol="https", host="gitlab.com"
-        )
+    def test_http_url_converts_http_to_https(self) -> None:
+        assert parse_extension_url("http://example.com/user/repo").remote_url.startswith("https://")
 
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_codeberg_url_sanitization(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        result = parse_extension_url("https://codeberg.org/user/repo/wiki")
-
-        assert result == UrlParseResult(
-            use_git=False,
-            url="https://codeberg.org/user/repo",
-            path="user/repo",
-            protocol="https",
-            host="codeberg.org",
+    def test_url_sanitization(self) -> None:
+        assert parse_extension_url("  https://example.com/path  ").remote_url == "https://example.com/path"
+        assert parse_extension_url("https://github.com/user/repo").remote_url == "https://github.com/user/repo.git"
+        assert parse_extension_url("git@gitlab.com:user/repo.git").remote_url == "https://gitlab.com/user/repo.git"
+        assert (
+            parse_extension_url("https://github.com/user/repo/blob/master").remote_url
+            == "https://github.com/user/repo.git"
```

Comment:
This test suite cover several different test methods from before (`test_codeberg_url_sanitization`, `test_github_url_sanitization`, `test_gitlab_url_sanitization`, `test_url_with_whitespace`, `test_ssh_url_reformatting`).